README.md

```diff
@@ -57,6 +57,11 @@ mavenCentral {
     // Token for Publisher API calls obtained from Sonatype official,
     // it should be Base64 encoded of "username:password".
     authToken = '<your token>'
+
+    // Whether the upload should be automatically published or not.
+    // Use `USER_MANAGED` if you wish to do this manually.
+    // This property is optional and defaults to `AUTOMATIC`.
+    publishingType = PublishingType.AUTOMATIC
```

Comment:
After my local testing, if we want to use the enumeration, we need to import them into gradle scripts. That's the same in your environment, right?

```groovy
import tech.yanand.gradle.PublishingType
```

If that's the case, then I recommend using string because it's a little less hassle for the user. At least I found that IntelliJ IDEA does not automatically prompt import.

I checked the official Gralde documentation and it doesn't seem to specifically recommend or discourage the use of enumerations in build scripts. But we don't seem to see much of it in normal times.

Also, if we implement it with string, we can validate the user's input to prevent illegal input. You can add a new exception to the `ExceptionFactory`.

Thanks again for your contributions!

Reply:
No problem - I have changed it to a String. You are right, this probably does make more sense in this case. Take a look at my changes to see if you are happy, I can make further adjustments if needed.